tests/field_defaults/tests.py

```diff
@@ -14,3 +33,217 @@ def test_field_defaults(self):
         self.assertIsInstance(a.id, int)
         self.assertEqual(a.headline, "Default headline")
         self.assertLess((now - a.pub_date).seconds, 5)
+
+    @skipUnlessDBFeature("can_return_columns_from_insert")
+    def test_field_database_defaults_returning(self):
+        a = DBArticle()
+        now = datetime.now()
+        a.save()
+
+        if connection.vendor == "sqlite":
+            # Round microseconds down to the nearest millisecond
+            microseconds = (now.microsecond // 1000) * 1000
+            now = now.replace(microsecond=microseconds)
```

Comment:
Can we use `connection.features.time_cast_precision` instead?
We're trying to avoid using `connection.vendor` in tests.